dimitriosv/week2/index.html

```diff
@@ -0,0 +1,316 @@
+                    totalMessages:4,
+                    inputListName:"",
+                    lists: [{listId: 1, listName: "first list1"},{listId: 2, listName: "second list2"}],
+                    allMessages: [  {messageId:1, belongsToList: 1, text: "testmessage1", IsArchived: false},
+                                    {messageId:2, belongsToList: 1, text: "testmessage2", IsArchived: false},
+                                    {messageId:3, belongsToList: 1, text: "testmessage3", IsArchived: false},
+                                    {messageId:4, belongsToList: 2, text: "testmessage4", IsArchived: false}]
+                };
+        },
+        eachList: function(list, i) {  
+        return (
+                <List key={i}
+                    index={i}
+                    listName={list.listName}
+                    listId={list.listId}
+                    allLists={this.state.lists}
+                    addMessageParent={this.addMessageParent}
+                    deleteMessageParent={this.deleteMessageParent}
+                    archiveMessageParent={this.archiveMessageParent}
+                    allMessages={this.state.allMessages}
+                >{list}</List>
+            );
+        },
+        createList: function(evt) {
+        if (this.state.inputListName=="") {
+            alert("Cannot add list with no name!")
+        } else {
+            var newlistId=this.state.totalLists
+            newlistId++; 
+            var newList = {
+                listId: newlistId,
+                listName: this.state.inputListName
+            };
+         
+            var lists=this.state.lists;
+             
+            lists.push(newList)
+            
+            this.setState({
+                  lists: lists,
+                  totalLists: this.state.totalLists+1
+            });
+        }
+             
+            
+        },
+        addMessageParent: function(listId,MessageToAdd) {
+
+            var newMessageId=this.state.totalMessages
+            newMessageId++; 
+
+
+            var newMessage = {
+                messageId:newMessageId,
+                belongsToList: listId,
+                text: MessageToAdd,
+                IsArchived: false
+            };
+         
+            var allMessages=this.state.allMessages;
+             
+            allMessages.push(newMessage)
+             
+            this.setState({
+                  allMessages: allMessages,
+                  totalMessages: this.state.totalMessages+1
+
+            });
+        },
+        deleteMessageParent: function(messageId) {
+            var allMessages=this.state.allMessages;
+
+            var newArray = allMessages.filter(function(obj) {
+                return messageId!=(obj.messageId) 
+            });
+                       
+            this.setState({
+                  allMessages: newArray
+            });
+             
+        },
+        archiveMessageParent: function(messageId) {
```

Comment:
I would name this `toggleArchivedMessageStatus` or something along those lines. _toggle_ is the keyword.